jwst/photom/photom.py

```diff
@@ -233,14 +240,14 @@ def calc_nirspec(self, ftab, area_fname):
                         sens2d[where_dq > 0] = 1.
                         self.input.dq = np.bitwise_or(self.input.dq, dqmap)
 
-                        # Divide the science data and uncertainty arrays by the
-                        # conversion factors
-                        self.input.data /= sens2d
-                        self.input.err /= sens2d
-                        self.input.var_poisson /= sens2d**2
-                        self.input.var_rnoise /= sens2d**2
+                        # Multiply the science data and uncertainty arrays by
+                        # the conversion factors
+                        self.input.data *= sens2d
+                        self.input.err *= sens2d
+                        self.input.var_poisson *= sens2d**2
+                        self.input.var_rnoise *= sens2d**2
                         if self.input.var_flat is not None and np.size(self.input.var_flat) > 0:
-                            self.input.var_flat /= sens2d**2
+                            self.input.var_flat *= sens2d**2
 
                         # Update BUNIT values for the science data and err
                         self.input.meta.bunit_data = 'mJy/arcsec^2'
```

Comment:
I think this may need updating to the new units, i.e. MJy or MJy/sr, depending on the source type.